golang/wasm/wrapper_js.go

```diff
@@ -210,15 +206,16 @@ func checkKeyStatus(_ js.Value, args []js.Value) any {
 }
 
 func findDOHEndpoint(_ js.Value, args []js.Value) any {
-	if len(args) != 1 {
-        // TODO: return rejected promise
-		return "expected 1 argument: domainName"
-	}
-	dohDomain := args[0].String()
+	handler := js.FuncOf(func(this js.Value, promises []js.Value) interface{} {
+		resolve := promises[0]
+		reject := promises[1]
 
-	handler := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
-		resolve := args[0]
-		reject := args[1]
+		if len(args) != 1 {
+			err := fmt.Errorf("expected 1 argument: domainName")
+			reject.Invoke(jsErr(err))
+			return nil
+		}
+		dohDomain := args[0].String()
```

Comment:
see my previous comment. With this construction, we always return a promise and reject it, making it possible to try{}catch{} or promise.catch(err => ...) properly